pkg/controller/nodes/dynamic/dynamic_workflow.go

```diff
@@ -38,18 +38,38 @@ func (d dynamicNodeTaskNodeHandler) buildDynamicWorkflowTemplate(ctx context.Con
 	}
 
 	currentAttemptStr := strconv.Itoa(int(nCtx.CurrentAttempt()))
+	uniqueParentID := nCtx.NodeStatus().GetUniqueNodeID()
```

Comment:
Something smells here (probably because of the existing code not your new additions)... If you look [here](https://github.com/lyft/flytepropeller/blob/2eae8a1312ce07caafba13eb4bef9a11ca27ebc9/pkg/controller/nodes/dynamic/dynamic_workflow.go#L123). `parentNodeStatus` is this "fake" node we create under the real node, we call it `dynamic-node` or something... so `nCtx.NodeStatus()` is not the same as `parentNodeStatus`... we should decide if we want to continue to have this fake node. I don't think it's required anymore because now we have support for direct node to node relationship... maybe :)

Reply:
@EngHabu I have commented there.

We have to until ParentTaskId is removed. That's not happening, until few versions out.